packages/module/src/DeleteModal/DeleteModal.tsx

```diff
@@ -0,0 +1,156 @@
+      itemName?: never;
+    };
+
+export type DeleteModalProps = Omit<ModalProps, 'children'> &
+  ItemProps & {
+    /** Delete variant. Destructive and extra-destructive variants will show a warning icon and danger button. For extra-destructive variant, text input confirmation is needed. */
+    deleteVariant?: 'extra-destructive' | 'destructive' | 'easily-recoverable';
+    /** Message describing what will deleted */
+    message?: {
+      toDelete?: string;
+      resourcesToDelete?: string;
+      endNote?: string;
+    };
+    /** Callback on clicking the delete button */
+    onDelete?: () => void;
+    /** Callback on clicking the cancel button */
+    onCancel?: () => void;
+    /** Error indicating deletion has failed */
+    error?: Error;
+    /** Modal ref */
+    ref: React.RefObject<Modal>;
+  };
+
+export const DeleteModal: React.FunctionComponent<DeleteModalProps> = ({
+  item,
+  items,
+  itemName,
+  itemNames,
+  deleteVariant = 'destructive',
+  onDelete,
+  onCancel,
+  message = {
+    resourcesToDelete: '',
+    endNote: ''
+  },
+  error,
+  ...props
+}: DeleteModalProps) => {
+  const isPlural = item === undefined;
+
+  const [confirmationText, setConfirmationText] = React.useState('');
+  const expectedConfirmationText = isPlural ? `delete ${itemNames.length} ${items}` : itemName;
+
+  const confirmed = deleteVariant === 'extra-destructive' ? confirmationText === expectedConfirmationText : true;
+
+  const toDeleteMessage = message.toDelete ?? ` and all of ${isPlural ? 'their' : 'its'} resources`;
+
+  const itemToDelete = `${itemNames ? `${itemNames.length} ` : ''}${item ?? items}`;
+  const modalHeaderTitle = `Delete ${itemToDelete}?`;
+
+  return (
+    <Modal variant="small" {...props}>
+      <ModalHeader
+        title={modalHeaderTitle}
+        titleIconVariant={deleteVariant !== 'easily-recoverable' ? 'warning' : undefined}
+      />
+      <ModalBody>
+        <Stack hasGutter>
+          <StackItem>
+            The{' '}
+            {isPlural ? (
+              `following ${items}`
+            ) : (
+              <>
+                <b>{itemName}</b> {item}
+              </>
+            )}
+            {toDeleteMessage}
+            {message.resourcesToDelete} will be deleted{isPlural && !message.endNote ? ':' : '.'} {message.endNote}
+            {isPlural && (
+              <List>
+                {itemNames.map((name) => (
+                  <ListItem key={name}>
+                    <b>{name}</b>
+                  </ListItem>
+                ))}
+              </List>
+            )}
+          </StackItem>
+          {deleteVariant === 'extra-destructive' && (
+            <StackItem>
+              <FormGroup
+                label={
+                  <span>
+                    Type <b>{expectedConfirmationText}</b> to confirm deletion:
+                  </span>
+                }
+              >
+                <TextInput
+                  aria-label="Delete modal input"
```

Comment:
Aria labels need to be configurable for internationalization purposes

Reply:
I'd recommend using the text that displays above the text input as the aria-label for this text input. This text visibly functions as the text input label, and it would be better to reference the visible text, rather than require consumers to specify this text in 2 places.

You can do this by providing an `id` on the div that wraps this text, and an `aria-labelledby` attribute on the `<TextInput>` component that references the `id` value for the text.

![image](https://github.com/user-attachments/assets/339d5e81-c4f6-433e-bd3a-898080b2192e)

But I do agree with @nicolethoen's comment above that this text  needs to be configurable:
> Type RedHatAwesome to confirm deletion: